tests/test_pydantic_utils.py

```diff
@@ -97,5 +99,42 @@ def test_dump_and_validate(self) -> None:
         self.assertIs(outer2d.inner.validated(override=4), outer2d.inner.validated(override=4))  # caching
 
 
+class SerializableExtensionsTestCase(unittest.TestCase):
+    """Tests for third-party types we add serializable annotations for."""
+
+    def test_region(self) -> None:
+        pixelization = Mq3cPixelization(10)
+        region = pixelization.pixel(12058823)
+        adapter = pydantic.TypeAdapter(SerializableRegion)
+        self.assertEqual(adapter.json_schema()["media"]["binaryEncoding"], "base16")
+        json_roundtripped = adapter.validate_json(adapter.dump_json(region))
+        self.assertIsInstance(json_roundtripped, ConvexPolygon)
+        self.assertEqual(json_roundtripped.getVertices(), region.getVertices())
+        python_roundtripped = adapter.validate_python(adapter.dump_python(region))
+        self.assertIsInstance(json_roundtripped, ConvexPolygon)
+        self.assertEqual(python_roundtripped.getVertices(), region.getVertices())
+        with self.assertRaises(ValueError):
+            adapter.validate_python(12)
+        with self.assertRaises(ValueError):
+            adapter.validate_json({})
+        with self.assertRaises(ValueError):
+            adapter.validate_json((b"this is not a region").hex())
+
```

Comment:
Might be worth adding a test for "a string, but not a hex string."  (Pretty sure it will pass already.)